### PR TITLE
[IMP] payment(_stripe),website_sale_delivery: change express checkout shipping heuristic

### PR DESCRIPTION
**Pull request by chevalierv**
Before this commit, when using express checkout, the shipping address was requested only if the technical module 'delivery' was installed.

Now, over express checkout, the shipping address will be requested if the sale order contains products that aren't services.

task-3149536

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale_delivery/controllers/main.py [hunk 1]
```diff
@@ -85,44 +85,45 @@ def process_express_checkout(self, shipping_address, shipping_option, **post):
         :param dict post: other information used in the parent route.
         :return int: The order's partner id.
         """
-        order_sudo = request.website.sale_get_order()
```
Comment (Demesmaeker) [inline addons/website_sale_delivery/controllers/main.py:88]: Maybe return when `not shipping_adress or not shipping option`, so you don't over indent everything... :thinking: 
Here you have a if in a if in a if, seems a bit too much. And this'll decreased the diff too =]

**Review by Demesmaeker — COMMENTED** [review]
Hello :duck:

**Review by Demesmaeker — APPROVED** [review]

File: addons/website_sale_delivery/controllers/main.py [hunk 2]
```diff
@@ -85,6 +85,9 @@ def process_express_checkout(self, shipping_address, shipping_option, **post):
         :param dict post: other information used in the parent route.
         :return int: The order's partner id.
         """
+        if not (shipping_address and shipping_option):
+            return super().process_express_checkout(**post)
```
Comment (AntoineVDV) [inline addons/website_sale_delivery/controllers/main.py:89]: The parent method doesn't seem to accept any kwarg.

File: addons/payment/views/payment_templates.xml [hunk 3]
```diff
@@ -323,6 +323,7 @@
               t-att-data-merchant-name="merchant_name"
               t-att-data-partner-id="partner_id"
               t-att-data-access-token="payment_access_token"
+              t-att-data-shipping-info-required="shipping_info_required"
```
Comment (AntoineVDV) [inline addons/payment/views/payment_templates.xml:326]: Update the comment above.

File: addons/payment/static/src/js/express_checkout_form.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ publicWidget.registry.PaymentExpressCheckoutForm = publicWidget.Widget.extend({
         await this._super(...arguments);
         this.txContext = {};
         Object.assign(this.txContext, this.$el.data());
+        this.txContext.shippingInfoRequired = !!this.txContext.shippingInfoRequired;
```
Comment (AntoineVDV) [inline addons/payment/static/src/js/express_checkout_form.js:16]: ES2019 restriction has been lifted 3 months too late :smiling_face_with_tear: 
`= this.txContextshippingInfoRequired ?? false`

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: addons/website_sale_delivery/controllers/main.py (hunk 2 above)
Comment (chevalierv) [reply addons/website_sale_delivery/controllers/main.py:89]: It's already like that below. When using the route, the arguments are all name params.

**Pull request closed without merging**